### PR TITLE
feat: Add Growthbook App

### PR DESCRIPTION
**Pull request by tzjames**
## Purpose

Growthbook is the #1 open-source & warehouse native experimentation platform.  This app lets Contentful customers create new experiments directly within Contentful and manage which content will be shown to the experiment's users.

## Approach

The app consists of a library to connect to the Growthbook API.  It is installed in three locations:

1. The Configuration Page: Here the installer can add in the Growthbook API key, host and other needed information to connect to the Growthbook API.  It also installs the Growthbook Experiment content type.
2. The Entry Editor: Here content managers can give a title to their experiment and add reference entries to the variations list.
3. The Sidebar: Here they can interact with the Growthbook API.  Once the entry has a title and two variations the "Create Experiment" button becomes valid, and clicking it will create an experiment on Growthbook. If any changes get made to the experiment an "Update Experiment" button is shown which can apply any changes.  Once everything is looking good they can click on a "Start Experiment" button which calls the Growthbook API to start the experiment.

Users of this app will have to install the Growthbook SDK on their site.  The Growthbook SDK will figure out which variation index to show to which users. Showing the correct content to the users for the Growthbook Experiment content type is as simple as selecting that same index in the variations list and looking up the content it references. 

## Testing steps

An [example app](https://github.com/growthbook/examples/pull/52) lays out the steps to integrate the Growthbook Contentful app to create a working site. 
https://github.com/growthbook/examples/pull/52
1. Clone the [growthbook examples repo](https://github.com/growthbook/examples).
2. Checkout the `ji/contentful` [branch](https://github.com/growthbook/examples/tree/ji/contentful).
3. cd contentful
4. Follow the instructions on the [readme](https://github.com/growthbook/examples/tree/ji/contentful/contentful).

Run `npm run tests`

## Breaking Changes
No it should not affect other apps.

## Dependencies and/or References

The [example app](https://github.com/growthbook/examples/pull/52) will be landed once this app is approved and on the marketplace.
The [wiki page](https://github.com/growthbook/growthbook/pull/3271) will also be published on Growthbook Docs.

## Deployment
The app can be installed by running `npm run dev` and installing a custom app on Contentful pointing to localhost:3000.

### CODE REVIEW

**Comment by david-shibley-contentful** [comment]
There is a failing test. Looks like we need to add a mock API key to the test and/or null check in the obfuscateApiKey method.

**Comment by tzjames** [comment]
> There is a failing test. Looks like we need to add a mock API key to the test and/or null check in the obfuscateApiKey method.
Thanks for pointing that out, it happened only after adding new obfuscation code.

File: apps/bynder-content-workflow/tsconfig.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
     "jsx": "react-jsx",
     "baseUrl": ".",
     "paths": {
-      "@/*": ["src/*"],
-    },
+      "@/*": ["src/*"]
```
Comment (mgoudy91) [inline apps/bynder-content-workflow/tsconfig.json:20]: Would you mind removing this (I'm assuming accidental) changes to the bynder content workflow app?

**Review by mgoudy91 — COMMENTED** [review]
Thanks for the submission! This is looking pretty good, just have a few questions for you about the app:

1. To run this locally, you are using `dev`? I tried using the `start` command, but it didn't appear to work in an expected way and couldn't find any specific instructions. The `dev` command did seem to work wrt sourcing the app locally on port 3000.
2. Curious if you were able to get a hosted bundle built with your `build` command to work. https://www.contentful.com/developers/docs/extensibility/app-framework/app-bundle/ Eventually this app will need to work as a hosted bundle, and I've been having problems getting this working with this app, but I could also be missing a step in your workflow.

Thanks for the help!

File: apps/bynder-content-workflow/tsconfig.json (hunk 1 above)
Comment (tzjames) [reply apps/bynder-content-workflow/tsconfig.json]: whoops done.

**Comment by tzjames** [comment]
> Thanks for the submission! This is looking pretty good, just have a few questions for you about the app:
> 
> 1. To run this locally, you are using `dev`? I tried using the `start` command, but it didn't appear to work in an expected way and couldn't find any specific instructions. The `dev` command did seem to work wrt sourcing the app locally on port 3000.

Thanks for taking a look.  Yes when I run `npm run build` it does successfully build to the directory /out.  Running `npm run start` does work for me then. Did you run `npm install`?

> 2. Curious if you were able to get a hosted bundle built with your `build` command to work. https://www.contentful.com/developers/docs/extensibility/app-framework/app-bundle/ Eventually this app will need to work as a hosted bundle, and I've been having problems getting this working with this app, but I could also be missing a step in your workflow.

I've been able to deploy it running `npm run upload` to https://app.contentful.com/account/organizations/6P3dU1el5fv4XYG0cThwq5/apps/definitions/7uAz1h1M4L5KgJszLnPG7f/general.

File: apps/growthbook/next.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  basePath: '',
+  assetPrefix: '',
```
Comment (mgoudy91) [inline apps/growthbook/next.config.js:4]:
```suggestion
  assetPrefix: './',
```
Looks like this was the change needed for our infrastructure to correctly serve the static assets. Let me know if this causes any issues on your end

**Comment by mgoudy91** [comment]
Perfect, thanks for the fixes and response. I took another look and identified the reason why we weren't able serve the statically hosted files. Let me know if that looks good to you and commit it if so.

After that we're ready to accept this PR and we'll get underway on the deployment process on our end

**Comment by tzjames** [comment]
> @tzjames thanks for making the requested change. We'll start the process of deploying the app today. 

That's great news. Let me know when it's deployed and we will land our docs change and example app.

**Comment by mgoudy91** [comment]
@tzjames it's out now, please let us know if you see any issues or problems with it working. It looked fine on our end, but we don't have a valid api key for growthbook so we can't really end-to-end test it.

Thanks again for the submission and your assistance with the process!

**Comment by tzjames** [comment]
@mgoudy91 Looks good.  It seems to be working.
